frontend/src/components/viewers/MetricsDropdown.tsx

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright 2022 The Kubeflow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { color, commonCss, fontsize, zIndex } from 'src/Css';
+import { classes, stylesheet } from 'typestyle';
+import { LinkedArtifact, getArtifactName } from 'src/mlmd/MlmdUtils';
+import TwoLevelDropdown, {
+  DropdownItem,
+  DropdownSubItem,
+  SelectedItem,
+} from 'src/components/TwoLevelDropdown';
+import {
+  ConfusionMatrixSection,
+  getHtmlViewerConfig,
+  getMarkdownViewerConfig,
+} from 'src/components/viewers/MetricsVisualizations';
+import PlotCard from 'src/components/PlotCard';
+import { ViewerConfig } from 'src/components/viewers/Viewer';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Banner from 'src/components/Banner';
+import { ExecutionArtifact, MetricsType, RunArtifact, SelectedArtifact } from 'src/pages/CompareV2';
+import { useQuery } from 'react-query';
+import { errorToMessage, logger } from 'src/lib/Utils';
+import { Execution } from 'src/third_party/mlmd';
+import { metricsTypeToString } from 'src/lib/v2/CompareUtils';
+
+const css = stylesheet({
+  leftCell: {
+    borderRight: `3px solid ${color.divider}`,
+  },
+  rightCell: {
+    borderLeft: `3px solid ${color.divider}`,
+  },
+  cell: {
+    borderCollapse: 'collapse',
+    padding: '1rem',
+    verticalAlign: 'top',
+  },
+  errorBanner: {
+    maxWidth: '40rem',
+  },
+  relativeContainer: {
+    position: 'relative',
+    height: '30rem',
+  },
+  visualizationPlaceholder: {
+    width: '40rem',
+    height: '30rem',
+    backgroundColor: color.lightGrey,
+    borderRadius: '1rem',
+    margin: '1rem 0',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  visualizationPlaceholderText: {
+    fontSize: fontsize.medium,
+    textAlign: 'center',
+    padding: '2rem',
+  },
+});
+
+interface MetricsDropdownProps {
+  filteredRunArtifacts: RunArtifact[];
+  metricsTab: MetricsType;
+  selectedArtifacts: SelectedArtifact[];
+  updateSelectedArtifacts: (selectedArtifacts: SelectedArtifact[]) => void;
+}
+
+export default function MetricsDropdown(props: MetricsDropdownProps) {
```

Comment:
Should I keep the `export default` keywords here, or place them at the bottom of the file as done in `CompareV2.tsx`? I moved the `MetricsDropdown` function up due to this comment: https://github.com/kubeflow/pipelines/pull/8026#discussion_r921436417.

Reply:
I think keeping the `export default` here looks good.